### ISSUE TITLE
Create a dependency-free browser implementation

### ISSUE THREAD

**Issue by Extarys** [ai/nanoid#102]
It would be awesome if we could create NanoId within the browser using the Crypto API

**Comment by ai** [comment]
> It would be nice to supplement it with the CryptoAPi or another PRNG as fall-back for the browser though.

It is a good philosophical question. In one hand it is cool to present “use the best available RPNG”. In another hand, RNG is **very** different. `Math.random()` is predictable, this is why there are a lot of cases when we will have security issues (or collision problems) on fallbacking to `Math.random()`.

To have a clear and predictable result (one method is always secure, another is always fast by non-secure) I split the API.

**Comment by ai** [comment]
If you do not need secure RNG, it will be always better to use `non-secure`, since it is much much more faster :D.

If you really have a case when you want to combine them, you can make your own wrap with `if ()` case.

Is this logic sound reasonable for you?

**Comment by Extarys** [comment]
Yes thanks you are probably right but I may try to tweak it to my needs. I'll let you know if I come with something clever :)
